filesystems/s3/src/main/scala/cromwell/filesystems/s3/S3PathBuilder.scala

```diff
@@ -167,6 +169,10 @@ case class S3Path private[s3](nioPath: NioPath,
 
   def key: String = safeAbsolutePath
 
+  lazy val s3Utils = new S3Utils()
+  lazy val s3Path = nioPath.asInstanceOf[org.lerch.s3fs.S3Path]
```

Comment:
By the time it gets here it's known to be a valid transformation. I'm not a huge fan of this and I **think** it can be alleviated directly in the underlying S3 code but this was a quicker win. I'm working out what can be done on the latter side

Reply:
Class cast issues are only exceptions, not instances of `java.lang.Error`, so it shouldn't threaten the VM. Depending on if the Filesystem was used in the call or the workflow level I'd expect either to fail, respectively.

Btw, same copypasta code, longer and slightly more scala-esque:

https://github.com/broadinstitute/cromwell/blob/f1cce2cd2723b849c4d8f285510f30913ec188a0/filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala#L221-L224

https://github.com/broadinstitute/cromwell/blob/f1cce2cd2723b849c4d8f285510f30913ec188a0/filesystems/oss/src/main/scala/cromwell/filesystems/oss/OssPathBuilder.scala#L131-L134

Reply:
So by nature of being the `nioPath` of a `cromwell.filesystems.s3.S3Path` it's already been determined that it's a valid `org.lerch.s3fs.S3Path`. 

There's a lot of redundancy in this whole relationship (in the "duplicated code" sense, not the "better safe than sorry" sense) and I'm working out the best feasible path. But since that potentially requires a PR against something other than Cromwell I want to get the less than ideal fix out now, as yes - saying things like "I know it can't happen, see the code 20 lines above" isn't something I like saying.

Reply:
That was my way of saying "I'd really prefer to leave it as is for now"